### PR TITLE
✨ :React student login

### PR DESCRIPTION
**Pull request by Charlie19255**
# Submitting Pull Request

### Ticket

ID number: 104

Branch name: react-student-login

Description: Configured App.js and studentlogin.js files to no longer automatically login to webpage. Have to sign in and send the appropriate email and password to rails side to login to the home page. Recieves the created JWT token as verification and logs both the email and password.

Link to ticket: https://www.notion.so/2022-Golf-Internship-a4556022acd84408a34062ba09f9e6f8?p=aadbed9474a84919964f13b8cb2e1d71&pm=s

### Notes

Notes for the reviewer: This was hard.

Contributors: Charlie, Sarah, Ramirez, Immanuel

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -10,19 +10,47 @@ import { CodeOfConduct } from "./pages/CodeOfConduct/CodeOfConduct"
...
 
   useEffect(() => {
     if (!login) {
-      navigate("/")
+      navigate("/login")
     }
   }, [login])
-
+  const verifyLogin = (field) => {
+    console.log(field)
```
Comment (sjproctor) [inline src/App.js:22]: Can you remove the log please!

File: src/App.js [hunk 2]
```diff
@@ -10,19 +10,47 @@ import { CodeOfConduct } from "./pages/CodeOfConduct/CodeOfConduct"
...
+      .then((response) => {
+        if (response.ok) {
+          // Code 200, navigate to home
+          setLogin(true)
+          navigate("/")
+        } else {
+          // Error code, log error message to console
+          console.log(response)
+          console.log(token)
+          console.error("Login failed")
```
Comment (sjproctor) [inline src/App.js:41]: There are logs here too.

File: src/pages/StudentLogin/StudentLogin.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Header } from "../../components/Header/Header"
 import { Checkbox, Label, TextInput } from "flowbite-react"
 
 const StudentLogin = (props) => {
-  const { login,setLogin,navigate } = props
+  const { login, navigate} = props
```
Comment (sjproctor) [inline src/pages/StudentLogin/StudentLogin.js:6]: Should have a space after navigate before the curly

File: src/pages/StudentLogin/StudentLogin.js [hunk 4]
```diff
@@ -17,16 +17,18 @@ const StudentLogin = (props) => {
       }
     )
   }
-
   const handleSubmit = (e) => {
+    const handleUserLoginObject={field}
```
Comment (sjproctor) [inline src/pages/StudentLogin/StudentLogin.js:21]: It looks like handleUserLoginObject only exists to pass the data down to verifyLogin
I think it makes sense to just pass the value directly in the verifyLogin and save a step.

**Review by sjproctor — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Charlie19255) [reply src/App.js]: Removed

File: src/App.js (hunk 2 above)
Comment (Charlie19255) [reply src/App.js]: Removed

File: src/pages/StudentLogin/StudentLogin.js (hunk 3 above)
Comment (Charlie19255) [reply src/pages/StudentLogin/StudentLogin.js]: Added

File: src/pages/StudentLogin/StudentLogin.js (hunk 4 above)
Comment (Charlie19255) [reply src/pages/StudentLogin/StudentLogin.js]: Removed handleUserLoginObject

**Review by sjproctor — APPROVED** [review]

**Pull request merged**